qiskit_nature/algorithms/pes_samplers/bopes_sampler.py

```diff
@@ -61,13 +68,15 @@ def __init__(
                 ``WindowExtrapolator``.
         """
 
-        self._gss = gss
+        self._solver_wrapper = gss
+        self._is_variational_solver = isinstance(self._solver_wrapper.solver, VariationalAlgorithm)
```

Comment:
I think this might fail if you supply e.g. a `VQEUCCFactory` to a `GroundStateSolver`. Please double check and fix accordingly

Reply:
Indeed, this _is_variational_solver needs to be defined after the get_qubit_operators() which has a line
```
if isinstance(self._solver, MinimumEigensolverFactory):
    # this must be called after transformation.transform
    self._solver = self._solver.get_solver(problem, self._qubit_converter)
```